File: java/rest-simple-exercise/src/main/java/com/example/restsimple/service/StudentServiceImpl.java
package com.example.restsimple.service;

import com.example.restsimple.model.Student;
import com.example.restsimple.repository.StudentRepository;
import com.example.restsimple.repository.UnitOfWork;
import com.example.restsimple.repository.UpdateStudent;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Service
public class StudentServiceImpl implements StudentService {

    private final StudentRepository studentRepository;
    private final UnitOfWork unitOfWork;

    @Autowired
    public StudentServiceImpl(StudentRepository studentRepository, UnitOfWork unitOfWork) {
        this.studentRepository = studentRepository;
        this.unitOfWork = unitOfWork;
    }

    @Override
    public List<Student> getAllStudents() {
        return new ArrayList<>();
    }

    @Override
    public Student createStudent(Student student) {
        return new Student();
    }

    @Override
    public Optional<Student> updateStudent(String id, UpdateStudent updatedStudent) {
        return Optional.empty();
    }

    @Override
    public Optional<Student> deleteStudent(String id) {
        return Optional.empty();
    }

    @Override
    public Optional<Student> findStudentById(String id) {
        return Optional.empty();
    }
}

File: kotlin/rest-simple-exercise/src/main/kotlin/com/example/restsimple/service/StudentServiceImpl.kt
package com.example.restsimple.service

import com.example.restsimple.model.Student
import com.example.restsimple.repository.StudentRepository
import com.example.restsimple.repository.UnitOfWork
import com.example.restsimple.repository.UpdateStudent
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class StudentServiceImpl @Autowired constructor(
    private val studentRepository: StudentRepository,
    private val unitOfWork: UnitOfWork
) : StudentService {

    override val allStudents: MutableList<Student?>
        get() {
            return mutableListOf()
        }

    override fun createStudent(student: Student): Student? {
        return Student()
    }

    override fun updateStudent(id: String?, updatedStudent: UpdateStudent): Optional<out Student?> {
        return Optional.empty<Student?>()
    }

    override fun deleteStudent(id: String?): Optional<out Student?> {
        return Optional.empty<Student?>()
    }

    override fun findStudentById(id: String?): Optional<Student> {
        return Optional.empty()
    }
}
